File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mp={}
        mp[0]=1
        result=sum=0
        for i in range(len(nums)):
            sum+=nums[i]
            rem = sum-k
            result+=mp.get(rem,0)
            mp[sum]=mp.get(sum,0)+1
        return result

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        for(int i:nums){
            mp[i]++;
        }
        vector<vector<int>>bucket(nums.size()+1);
        for(auto [num,freq]:mp){
            bucket[freq].push_back(num);
        }
        vector<int>result;
        for(int i=nums.size();i>0 && result.size()<k;i--){
            for(int num : bucket[i]){
                result.push_back(num);
                if(result.size()==k) break;
            }
        }
        return result;
    }
};
